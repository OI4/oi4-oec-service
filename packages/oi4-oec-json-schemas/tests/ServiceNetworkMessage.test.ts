import { matchersWithOptions } from 'jest-json-schema'
import schema from '../src/schemas/ServiceNetworkMessage.schema.json'
import serviceParameterRequest from '../src/schemas/ServiceParametersRequest.schema.json'
import serviceParameterResponse from '../src/schemas/ServiceParametersResponse.schema.json'
import callMethodRequest from '../src/schemas/CallMethodRequest.schema.json'
import callMethodResult from '../src/schemas/CallMethodResult.schema.json'
import baseDataType from '../src/schemas/BaseDataType.schema.json'
import uint32 from '../src/schemas/dataTypes/uint32.schema.json'

import validObjs from './__fixtures__/ServiceNetworkMessage_valid.json'
import invalidObjs from './__fixtures__/ServiceNetworkMessage_invalid.json'


expect.extend(
  matchersWithOptions({
    // Loading in a schema which is comprised only of definitions,
    // which means specific test schemas need to be created.
    // This is good for testing specific conditions for definition schemas.
    schemas: [serviceParameterRequest, serviceParameterResponse,
    callMethodRequest, callMethodResult, baseDataType, uint32],
    verbose: true,
  })
)

describe('ServiceNetworkMessage schema', () => {
    it('validate schema', () => {
      expect(schema).toBeValidSchema()
    })
  
    it.each(validObjs as [])(
      '(%#) match valid config object to schema -> %s',
      (_name: string, obj) => {
        expect(obj).toMatchSchema(schema)
      }
    )
  
    it.each(invalidObjs as [])(
      '(%#) match fails for invalid config -> %s',
      (name: string, obj) => {
        expect(obj).not.toMatchSchema(schema)
        expect(() => {
          expect(obj).toMatchSchema(schema)
        }).toThrowErrorMatchingSnapshot(name)
      }
    )
  })