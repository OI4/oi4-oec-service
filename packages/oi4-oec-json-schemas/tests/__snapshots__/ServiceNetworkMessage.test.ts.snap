// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceNetworkMessage schema (0) match fails for invalid config -> empty: empty 1`] = `
"expect(received).toMatchSchema(schema)

received
  must have required property 'MessageId'
    Path:     ServiceNetworkMessage.schema.json#/required
  must have required property 'MessageType'
    Path:     ServiceNetworkMessage.schema.json#/required
  must have required property 'PublisherId'
    Path:     ServiceNetworkMessage.schema.json#/required
  must have required property 'Message'
    Path:     ServiceNetworkMessage.schema.json#/required
"
`;

exports[`ServiceNetworkMessage schema (1) match fails for invalid config -> Missing methodId in Request: Missing methodId in Request 1`] = `
"expect(received).toMatchSchema(schema)

received
  /Message must have required property 'MethodsToCall'
    Received: <object> {\\"methodsToCall\\":[{\\"inputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'methodsToCall'
    Expected: <boolean> false
    Received: <object> {\\"methodsToCall\\":[{\\"inputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must have required property 'Results'
    Received: <object> {\\"methodsToCall\\":[{\\"inputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'methodsToCall'
    Expected: <boolean> false
    Received: <object> {\\"methodsToCall\\":[{\\"inputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must match exactly one schema in oneOf
    Expected: [{\\"$ref\\":\\"ServiceParametersRequest.schema.json\\"},{\\"$ref\\":\\"ServiceParametersResponse.schema.json\\"}]
    Received: <object> {\\"methodsToCall\\":[{\\"inputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/properties/Message/oneOf
"
`;

exports[`ServiceNetworkMessage schema (2) match fails for invalid config -> Missing inputArguments in Request: Missing inputArguments in Request 1`] = `
"expect(received).toMatchSchema(schema)

received
  /Message must have required property 'MethodsToCall'
    Received: <object> {\\"methodsToCall\\":[{\\"methodId\\":\\"newDataSetWriterId\\"}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'methodsToCall'
    Expected: <boolean> false
    Received: <object> {\\"methodsToCall\\":[{\\"methodId\\":\\"newDataSetWriterId\\"}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must have required property 'Results'
    Received: <object> {\\"methodsToCall\\":[{\\"methodId\\":\\"newDataSetWriterId\\"}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'methodsToCall'
    Expected: <boolean> false
    Received: <object> {\\"methodsToCall\\":[{\\"methodId\\":\\"newDataSetWriterId\\"}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must match exactly one schema in oneOf
    Expected: [{\\"$ref\\":\\"ServiceParametersRequest.schema.json\\"},{\\"$ref\\":\\"ServiceParametersResponse.schema.json\\"}]
    Received: <object> {\\"methodsToCall\\":[{\\"methodId\\":\\"newDataSetWriterId\\"}]}
    Path:     ServiceNetworkMessage.schema.json#/properties/Message/oneOf
"
`;

exports[`ServiceNetworkMessage schema (3) match fails for invalid config -> Missing statusCode in response : Missing statusCode in response  1`] = `
"expect(received).toMatchSchema(schema)

received
  /Message must have required property 'MethodsToCall'
    Received: <object> {\\"results\\":[{\\"inputArgumentResults\\":[0],\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"inputArgumentResults\\":[0],\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must have required property 'Results'
    Received: <object> {\\"results\\":[{\\"inputArgumentResults\\":[0],\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"inputArgumentResults\\":[0],\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must match exactly one schema in oneOf
    Expected: [{\\"$ref\\":\\"ServiceParametersRequest.schema.json\\"},{\\"$ref\\":\\"ServiceParametersResponse.schema.json\\"}]
    Received: <object> {\\"results\\":[{\\"inputArgumentResults\\":[0],\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/properties/Message/oneOf
"
`;

exports[`ServiceNetworkMessage schema (4) match fails for invalid config -> Missing inputArgumentResults in response: Missing inputArgumentResults in response 1`] = `
"expect(received).toMatchSchema(schema)

received
  /Message must have required property 'MethodsToCall'
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must have required property 'Results'
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must match exactly one schema in oneOf
    Expected: [{\\"$ref\\":\\"ServiceParametersRequest.schema.json\\"},{\\"$ref\\":\\"ServiceParametersResponse.schema.json\\"}]
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"outputArguments\\":[{\\"resource\\":\\"referenceDesignation\\"}]}]}
    Path:     ServiceNetworkMessage.schema.json#/properties/Message/oneOf
"
`;

exports[`ServiceNetworkMessage schema (5) match fails for invalid config -> Missing outputArguments in response: Missing outputArguments in response 1`] = `
"expect(received).toMatchSchema(schema)

received
  /Message must have required property 'MethodsToCall'
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"inputArgumentResults\\":[0]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"inputArgumentResults\\":[0]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must have required property 'Results'
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"inputArgumentResults\\":[0]}]}
    Path:     ServiceNetworkMessage.schema.json#/required
  must NOT have additional properties, but found 'results'
    Expected: <boolean> false
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"inputArgumentResults\\":[0]}]}
    Path:     ServiceNetworkMessage.schema.json#/additionalProperties
  /Message must match exactly one schema in oneOf
    Expected: [{\\"$ref\\":\\"ServiceParametersRequest.schema.json\\"},{\\"$ref\\":\\"ServiceParametersResponse.schema.json\\"}]
    Received: <object> {\\"results\\":[{\\"statusCode\\":0,\\"inputArgumentResults\\":[0]}]}
    Path:     ServiceNetworkMessage.schema.json#/properties/Message/oneOf
"
`;
